### ISSUE TITLE
Notice: accessibility improvements and fixes

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#54497]
### Description

The Nocice component accessibility needs fome improvements and fixes:

1
When screen reader users navigat econtent and find a notice, there's nothing to inform that this text is a 'notice'. For example with this notice in the Site editor:

<img width="381" alt="Screenshot 2023-09-15 at 12 22 13" src="https://github.com/WordPress/gutenberg/assets/1682452/4a7c2b40-88bc-484c-b5f0-6ec87be122fd">

Screen reaer users will only hear the text content `Looking for template parts? Find them in "Patterns".`
Only when moving to the Close button they will hear `Dismiss thie notice` (more on this later). No other reference to the term 'notice' is provided.
As such, therre is no parity between the information provided visually and the inforation provided semantically.

2
There are 4 types of notice: `'warning' | 'success' | 'error' | 'info'`. They are only communicated by the means of the left border color. respectively: a yellow-orangish, green, red, or blue border.

<img width="381" alt="Screenshot 2023-09-15 at 12 22 13" src="https://github.com/WordPress/gutenberg/assets/1682452/9e9340f3-edf2-45d5-959c-83f40c8a849f">
<img width="356" alt="Screenshot 2023-09-15 at 13 08 02" src="https://github.com/WordPress/gutenberg/assets/1682452/9ad01602-a064-49bf-baca-4c8f92703d9e">
<img width="356" alt="Screenshot 2023-09-15 at 13 08 55" src="https://github.com/WordPress/gutenberg/assets/1682452/9c26c364-7da8-4f2a-9210-bbc5a365b069">
<img width="354" alt="Screenshot 2023-09-15 at 13 09 48" src="https://github.com/WordPress/gutenberg/assets/1682452/6b4e1b2b-77a7-4031-b425-4d92c92a9f67">

As such, also visuallly this is not accessible because users with color perception impairment would not be able to distinguish the type of notice. Ideally, there should be some visible text to communicat ethe type. This point would require design feedback. Ideaelly, I would like to see the notice type provided by teh means of some visible text, for example:

<img width="358" alt="Screenshot 2023-09-15 at 14 58 17" src="https://github.com/WordPress/gutenberg/assets/1682452/47a1615c-0f1d-4b67-a750-ce23d3d8d798">

At the very least, there should be some visually hidden text.

3
The X close button dosn't have a tooltip. This is a bug. s explained several time across the history of this project, **buttons that only use an icon muyst visually expose their accessible name in some way**. We do use tooltips for this purpose, and worth reminding that tooltips are already a compromise. Ideally, buttons should use visible text. At the very least icon buttons must use a tooltip. Looking at the code, the lack of a tooltip is intentional, as [showTooltip={ false }](https://github.com/WordPress/gutenberg/blame/8a43c8582c88c853ec12433a1c5125ebcdf42993/packages/components/src/notice/index.tsx#L159) is set on the button. 
It appears that trying to educate developers on this point seems not enough, as this is a very simple accessibility requirements but still another case of a button with no tooltip. At this point, I'd like to see the tooltip enforced via code for buttons that only use icons. Cc @ciampo 

4
The X close button label is `Dismiss this notice`. While this is the accessible name traditionally used in WordPress core, i don't think it is ideal. As a speech recongition softare user I see a X icon. As such, I'd llikely try to use the following voice commands:
- Click X
- Click close

These voice commands would fail because the actual accessible name of the button is `Dismiss this notice`. The root problem heere is that the icon suggest the concept of 'Close', while the actual name uses the term 'dismiss'.  The mismatch between the visual represented action and the accessible name would prevent speech recognition software users from being able to use this control

5
Lastly, a small technical detail. The close button callback users `event.preventDefault()`, not sure why. This button does have a HTML attribute `type="button"`. Buttons with `type="button"` do not have any default action to prevent, not even when places inside a form. Seems to me `event.preventDefault()` is redundant, unless I'm missing something.

### Step-by-step reproduction instructions

- Use a clean database or make sure the preference `isTemplatePartMoveHintVisible` in the database wp_usermeta > wp_persisted_preferences is set to 1.
- Go to the side editor
- In the navigation panel observe the `Looking for template parts? Find them in "Patterns".` notice is rendered.
- Inspect the source and observe there's no information on what this text is (a notice).
- Inspect the source and observe there's no textual information on the notice type ('warning' | 'success' | 'error' | 'info')
- Hover or focus the X close button and observe no tooltip is rendered

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
